Add documentation section regarding AI assistance (#721)
vllm-project/guidellm: .github/PULL_REQUEST_TEMPLATE.md

## Use of AI

- [ ] Includes code generated or substantially modified by an AI agent
- [ ] Includes tests generated or substantially modified by an AI agent

> NOTE: the `Generated-by` and `Assisted-by` tags should be used in both git commit messages and docstrings to indicate when code or tests were generated or substantially modified by an AI agent, as described in the project's [`DEVELOPING.md`](https://github.com/vllm-project/guidellm/blob/main/DEVELOPING.md) file.

Reviewer: ```suggestion
> NOTE: the `Generated-by` and `Assisted-by` tags should be used in both git commit messages and docstrings to indicate when code or tests were generated or substantially modified by an AI agent, as described in the project's [`DEVELOPING.md`](https://github.com/vllm-project/guidellm/blob/main/DEVELOPING.md) file.
```

Author: I was tempted to use `> [!NOTE]\` or even `> [!IMPORTANT]\` ... except that while the template is markdown and it'd presumably render, in the end this becomes a git commit description which won't do any formatting. But the `>` won't obfuscate the commit description and does set off the note a bit in the template -- so, sure.

Reviewer: It also does not work in PRs. GitHub only supports that format for issues and in-repo docs.

> > NOTE: the `Generated-by` and `Assisted-by` tags should be used in both git commit messages and docstrings to indicate when code or tests were generated or substantially modified by an AI agent, as described in the project's [`DEVELOPING.md`](https://github.com/vllm-project/guidellm/blob/main/DEVELOPING.md) file.

Reviewer: ```suggestion
> NOTE: the `Generated-by` and `Assisted-by` tags should be used in git commit messages to indicate when code is generated or substantially modified by an AI agent, as described in the project's [`DEVELOPING.md`](https://github.com/vllm-project/guidellm/blob/main/DEVELOPING.md) file.
```

Keep it simple

Author: Dang -- I forgot to update the template file.
